core-backend/src/funcs.rs

```diff
@@ -592,54 +584,55 @@ where
         )
     }
 
-    pub fn read(at: u32, len: u32, buffer_ptr: u32) -> impl Syscall<Caller> {
-        FallibleSyscall::new::<ErrorBytes>(CostToken::Read, move |ctx: &mut CallerWrap<Caller>| {
-            let payload_lock = ctx.ext_mut().lock_payload(at, len)?;
-            payload_lock
-                .drop_with::<MemoryAccessError, _>(|payload_access| {
-                    let mut f = || {
-                        let mut registry = MemoryAccessRegistry::default();
-                        let write_buffer = registry.register_write(buffer_ptr, len);
-                        let mut io = registry.pre_process(ctx)?;
-                        io.write(ctx, write_buffer, payload_access.as_slice())?;
-                        Ok(())
-                    };
-                    let res = f();
-                    let unlock_bound = ctx.ext_mut().unlock_payload(payload_access.into_lock());
-
-                    DropPayloadLockBound::from((unlock_bound, res))
-                })
-                .into_inner()
-                .map_err(Into::into)
-        })
+    pub fn read(at: u32, buffer: Write) -> impl Syscall<Caller> {
```

Comment:
isn't it a bug, changing order of args?

Reply:
Yeah, but no. The arguments are in reverse order, but there is no other way but to change the syscall signature, which is obviously not worth it.

```
let ptr = SyscallValue(args[1]).try_into()?;
let size = SyscallValue(args[0]).try_into()?;
```